test/release/examples/primers/specialMethods.chpl

```diff
@@ -0,0 +1,172 @@
+// operations such as initializing or deinitializing a class or record
+// instance, accessing a class or record as though it were an array,
+// iterating over an class or record, and defining the way a class or record
+// is read from or written to a channel.
+//
+// Note that there are two ways to declare a method. Methods declared within
+// a class or record are called primary methods:
+
+record ExampleRecord1 {
+  var exampleField: int;
+  proc primaryMethod() { }
+}
+
+
+// Methods declared outside of a class or record are called secondary methods:
+
+record ExampleRecord2 {
+  var exampleField: int;
+}
+proc ExampleRecord2.secondaryMethod() { }
+
+// This primer will use the secondary method form, but all of the special
+// methods can also be written as primary methods.
+
+// First we will declare a simple record with a field that is a tuple of
+// integers.  We'll add special methods and iterators to this record later.
+record R {
+  param size: int = 10;
+  var vals: size*int;
+}
+
+/*
+  Initializers and Deinitializers
+  -------------------------------
+*/
+
+// An initializer is called when creating an instance of the class or record,
+// for example with the ``new`` keyword. An initializer accepting zero
+// arguments is called a *default initializer*.  An initializer that takes
+// another instance of the same class or record is called a *copy initializer*.
+
+// The ``deinit`` method will deinitialize a record when it leaves scope,
+// or a class when ``delete`` is called on it. If the class or record
+// contained any unmanaged classes, open files, etc. this method would be
+// the place to delete them or otherwise clean them up. See the :doc:`records`
+// primer for more details on initializers and deinitializers.
+
+/*
+  The ``this`` Accessor
+  ---------------------
+*/
+
+// The ``this`` method gives the record the ability to be accessed like an
+// array.  Here we use the index to choose a tuple element.
+proc R.this(n: int) ref {
+  if n < 1 || n > size then
+    halt("index out of bounds accessing R");
+  return vals[n];
+}
+
+var r = new R();
+
+// All functions and methods in Chapel can be called using either parenthesis
+// or square brackets. Here we use square brackets to style it as an array
+// access.
+r[1] = 1;
+r[3] = 3;
+
+writeln(r.vals);
+
+/*
+  Default Iterators
+  -----------------
+*/
+
+// An iterator named ``these`` is automatically called when a record or class
+// instance is used in the iterator position of a ``for`` loop.
+iter R.these() ref {
+  for i in 1..size {
+    yield vals[i];
+  }
+}
+
+for val in r {
+  val += 1;
+}
+writeln(r.vals);
+
+// Classes and records can also define parallel iterators including
+// leader/follower iterator pairs and standalone parallel iterators. For
+// more information on parallel iterators, see the :doc:`parIters` primer.
+
+/*
+  IO Methods
+  ----------
+*/
+
+// Define how to write an instance of R to a channel. We'll write the
```

Comment:
I think most of the text in this primer has been complete sentences, but "Define how to write...." isn't.